playground.js

```diff
@@ -0,0 +1,278 @@
+    typeLabel: '{underline ID}',
+    type: String
+  },
+  {
+    name: 'user',
+    description: 'The email of the user to log in with. The SDK is initialized in an authenticated state.',
+    alias: 'u',
+    typeLabel: '{underline email}',
+    type: String
+  },
+  {
+    name: 'password',
+    description: 'The password of the user to log in with. The SDK is initialized in an authenticated state.',
+    alias: 's',
+    typeLabel: '{underline password}',
+    type: String
+  },
+  {
+    name: 'raw',
+    description: 'Start the playground without initializing the SDK.',
+    type: Boolean
+  },
+  {
+    name: 'script',
+    description: 'Execute a playground script, i.e. a file that contains one or more playground commands.',
+    typeLabel: '{underline script_file.js}',
+    type: String
+  },
+  {
+    name: 'apidocs',
+    description: 'Open the Marketplace API reference using default browser.',
+    type: Boolean
+  }
+];
+
+const printUsage = () => {
+  console.log(commandLineUsage([
+    {
+      header: 'Marketplace API Playground',
+      content: 'The easiest way to call the Flex Marketplace API using the JS SDK to try things out, to learn the APIs and to test your ideas.'
+    },
+    {
+      header: 'Options',
+      optionList: optionDefinitions
+    }
+  ]));
+};
+
+const printResponse = response => {
+  console.log(JSON.stringify(response.data, null, 4));
+  return response;
+}
+
+const apiDocs = () =>
+      open('https://www.sharetribe.com/api-reference/marketplace.html?javascript#marketplace-api')
+      .then(() => '');
+
+// Parse command line args
+//
+const options = commandLineArgs(optionDefinitions);
+
+if (options.apidocs) {
+  apiDocs();
+  process.exit();
+}
+
+if (options.help || (!options.raw && options.clientid == null)) {
+  printUsage();
+  process.exit();
+};
+
+let scriptSrc;
+if (options.script) {
+  scriptSrc = fs.createReadStream(options.script);
+} else {
+  scriptSrc = null;
+}
+
+const setupContext = (ctx, sdk) => {
+  ctx.sharetribeSdk = sharetribeSdk;
+  if (sdk) {
+    ctx.sdk = sdk;
+  }
+  ctx.printResponse = printResponse;
+  ctx.apiDocs = apiDocs;
+  ctx.LatLng = LatLng;
+  ctx.LatLngBounds = LatLgnBounds;
+  ctx.UUID = UUID;
+  ctx.Money = Money;
+  ctx.BigDecimal = BigDecimal;
+}
+
+const startRepl = (sdk, rawMode, src) =>
+      () => {
+        if (src) {
+          // Setup context
+          setupContext(global, sdk);
+
+          // Start REPL
+          console.log('Executing script...');
+          console.log();
+          repl.start({
+            prompt: '> ',
+            input: src,
+            output: process.stdout,
+            useGlobal: true,
+            writer: result => result instanceof Promise ? '' : util.inspect(result)
+          });
+        } else {
+          printWelcomeMessage(rawMode);
+          // Start REPL
+          const replInstance = repl.start('> ');
+
+          // Attach history
+          replInstance.setupHistory('./.repl_history', () => null);
```

Comment:
Change request: I got an error because my Node version was too old for this. Mine was 10.11.0, and this was added in 11.10.0.

We don't specify the minimum required Node version in package.json. We could do that, or we could just use the old repl history library.

Reply:
The history package we used before has an inverse problem; it's using now deprecated node internals and will stop working (or maybe even has stopped working with the very latest node). It seems to be an unmaintained lib. There's an issue in there but it's been unaddressed for longer time and last commit is from several years ago. So I'll go with adding a minimum node version required.

Reply:
IRL: We can detect this and opt out of using it if node version is old enough to not have it. No need to add min version because of this.